src/index.js

```diff
@@ -20,33 +31,125 @@ const toDoTasks = [
   },
 ];
 
-const createItemTask = (task) => {
-  if (task.completed === false) {
-    listContainer.insertAdjacentHTML('beforeend',
-      `<li>
-    <input type="checkbox" name="task${task.index}" value="task${task.index}">
-    <label class="text-task">${task.description}</label><br>
-    <i class="fas fa-ellipsis-v icon-item"></i>
-    </li>
-    `);
+const displayTasks = () => {
+  listContainer.innerHTML = '';
+  toDoTasks.forEach((task) => createItemTask(task));
+};
+
+const refrestCompletedTask = (bool, element) => {
+  if (bool) {
+    toDoTasks[parseInt(element.dataset.id, 10)].completed = true;
+    element.setAttribute('checked', 'true');
   } else {
-    listContainer.insertAdjacentHTML(
-      'beforeend',
-      `<li>
-    <input type="checkbox" name="task${task.index}" value="task${task.index}" checked>
-    <label class="text-task">${task.description}</label><br>
-    <i class="fas fa-ellipsis-v icon-item"></i>
-    </li>
-    `,
-    );
+    toDoTasks[parseInt(element.dataset.id, 10)].completed = false;
+    element.removeAttribute('checked');
   }
 };
 
-const displayTasks = () => {
-  listContainer.innerHTML = '';
-  toDoTasks.forEach((task) => createItemTask(task));
+const getTaskData = () => {
+  if (localStorage.getItem('TaskData') != null) {
+    toDoTasks = JSON.parse(localStorage.getItem('TaskData'));
+  }
+};
+
+const setData = () => {
+  localStorage.setItem('TaskData', JSON.stringify(toDoTasks));
+};
+
+const refrestTargetDragDrop = () => {
+  const tasks = document.querySelectorAll('.container-list .item');
+
+  tasks.forEach((task) => {
+    task.addEventListener('dragstart', dragStart);
+    task.addEventListener('dragend', dragEnd);
+    task.addEventListener('dragover', dragOver);
+    task.addEventListener('drop', (e) => {
+      dragDrop(e);
+      const items = [...listContainer.children];
+
+      toDoTasks = [];
+
+      items.forEach((item, index) => {
+        const newTask = new Task(item.children[1].textContent, item.children[0].checked, index);
+        toDoTasks.push(newTask);
+        setData();
+      });
```

Comment:
- [ ] By default, new tasks should have the property index set to the value of the new array length (i.e. if a 4th task is added to the list, the index of that task should equal to 5 or the index of the 1rst task should equal to 1), the index start by 1,2,... and not 0,1,2,... Please kindly fix that 🙃 😉

Reply:
Thank you, I will update soon.